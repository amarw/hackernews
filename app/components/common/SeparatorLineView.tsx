import React from 'react';
import { View, StyleSheet, StyleProp, ViewStyle } from 'react-native';
import Color from 'app/utilities/Color';

export interface SeparatorLineViewProps {
  thickness?: number;
  color?: string;
  vertical?: boolean;
  style?: StyleProp<ViewStyle>;
}

const SeparatorLineView: React.FC<SeparatorLineViewProps> = ({
  thickness = StyleSheet.hairlineWidth,
  color = Color.border,
  vertical = false,
  style,
  ...otherProps
}: SeparatorLineViewProps) => {
  const modifiedStyle = [
    style,
    {
      width: vertical ? thickness : undefined, // for vertical line thickness is width
      height: vertical ? undefined : thickness, // for horizontal line thickness is height
      backgroundColor: color
    }
  ];
  return <View style={modifiedStyle} {...otherProps} />;
};
export default SeparatorLineView;
