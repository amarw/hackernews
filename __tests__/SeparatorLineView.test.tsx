import 'react-native';
import * as React from 'react';
import SeparatorLineView from "../app/components/common/SeparatorLineView";

import renderer from 'react-test-renderer';

test('renders horizontal SeparatorLineView correctly', () => {
    const tree = renderer.create(<SeparatorLineView />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders horizontal SeparatorLineView with custom props correctly', () => {
    const tree = renderer.create(<SeparatorLineView thickness={10} color={'#000000'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders horizontal SeparatorLineView correctly', () => {
    const tree = renderer.create(<SeparatorLineView vertical />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders vertical SeparatorLineView with custom props correctly', () => {
    const tree = renderer.create(<SeparatorLineView vertical thickness={10} color={'#000000'} />).toJSON();
    expect(tree).toMatchSnapshot();
});