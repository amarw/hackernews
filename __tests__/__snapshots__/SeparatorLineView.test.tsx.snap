// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders horizontal SeparatorLineView correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#cccccc",
        "height": 0.5,
        "width": undefined,
      },
    ]
  }
/>
`;

exports[`renders horizontal SeparatorLineView correctly 2`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#cccccc",
        "height": undefined,
        "width": 0.5,
      },
    ]
  }
/>
`;

exports[`renders horizontal SeparatorLineView with custom props correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#000000",
        "height": 10,
        "width": undefined,
      },
    ]
  }
/>
`;

exports[`renders vertical SeparatorLineView with custom props correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#000000",
        "height": undefined,
        "width": 10,
      },
    ]
  }
/>
`;
